From af203a7a3ef6e03f0e2cf40b7d2049ff54f13d0c Mon Sep 17 00:00:00 2001
From: Alon Levy <alevy@redhat.com>
Date: Fri, 18 Jan 2013 06:11:08 +0100
Subject: [PATCH 1/5] qxl: fix range check for rev3 io commands.

RH-Author: Alon Levy <alevy@redhat.com>
Message-id: <1358489468-22362-1-git-send-email-alevy@redhat.com>
Patchwork-id: 47570
O-Subject: [PATCH RHEL-6.4 qemu-kvm] qxl: fix range check for rev3 io commands.
Bugzilla: 876982
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>
RH-Acked-by: Amos Kong <akong@redhat.com>

From: Gerd Hoffmann <kraxel@redhat.com>

Enables QXL_IO_FLUSH_SURFACES_ASYNC and QXL_IO_FLUSH_RELEASE
which are part of the qxl rev3 feature set.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
(cherry picked from commit ffe01e599f106bedf13ba0e0f6956735b1942a0c)

RHBZ: 876982
---
 hw/qxl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/qxl.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hw/qxl.c b/hw/qxl.c
index 84b1398..89d855a 100644
--- a/hw/qxl.c
+++ b/hw/qxl.c
@@ -1379,7 +1379,7 @@ static void ioport_write(void *opaque, uint32_t addr, uint32_t val)
     }
 
     if (d->revision <= QXL_REVISION_STABLE_V10 &&
-        io_port >= QXL_IO_FLUSH_SURFACES_ASYNC) {
+        io_port > QXL_IO_FLUSH_RELEASE) {
         qxl_set_guest_bug(d, "unsupported io %d for revision %d\n",
             io_port, d->revision);
         return;
-- 
1.7.1

